client/packages/inventory/src/Stocktake/ListView/CreateStocktakeModal.tsx

```diff
@@ -5,98 +5,133 @@ import {
-    useState<CreateStocktakeInput>(defaultCreateStocktakeInput);
-  const [selectedLocation, setSelectedLocation] =
-    useState<LocationRowFragment | null>(null);
-  const [selectedMasterList, setSelectedMasterList] =
-    useState<MasterListRowFragment | null>(null);
+  const [{ location, masterList, expiryDate, createBlankStocktake }, setState] =
+    useState<ModalState>({
+      location: null,
+      masterList: null,
+      expiryDate: null,
+      createBlankStocktake: false,
+    });
+
+  const stockFilter: StockLineFilterInput = {
+    location: location
+      ? {
+          id: { equalTo: location.id },
+        }
+      : null,
+    masterList: masterList
+      ? {
+          id: { equalTo: masterList.id },
+        }
+      : null,
+    expiryDate: expiryDate
+      ? { beforeOrEqualTo: Formatter.naiveDate(expiryDate) }
+      : null,
+    hasPacksInStore: true,
+  };
+
+  const { data } = useStockListCount(stockFilter);
 
   const { localisedDate } = useFormatDateTime();
 
   const generateComment = () => {
-    const { locationId, masterListId, itemsHaveStock, expiresBefore } =
-      createStocktakeArgs;
-    if (masterListId && selectedMasterList) {
-      return t('stocktake.comment-list-template', {
-        list: selectedMasterList.name,
-      });
-    }
+    if (createBlankStocktake) return '';
 
-    if (locationId && selectedLocation) {
-      return t('stocktake.comment-location-template', {
-        location: selectedLocation.code,
-      });
-    }
+    const filterComments: string[] = [];
 
-    if (itemsHaveStock) {
-      return t('stocktake-comment-items-have-stock-template');
+    if (!!masterList) {
+      filterComments.push(
+        t('stocktake.master-list-template', {
+          masterList: masterList.name,
+        })
+      );
+    }
+    if (!!location) {
+      filterComments.push(
+        t('stocktake.location-template', {
+          location: location.code,
+        })
+      );
     }
-    if (expiresBefore) {
-      return t('stocktake.comment-expires-before-template', {
-        date: localisedDate(expiresBefore),
-      });
+    if (!!expiryDate) {
+      filterComments.push(
+        t('stocktake.expires-before-template', {
+          date: localisedDate(expiryDate),
+        })
+      );
     }
+
+    if (filterComments.length === 0) return undefined;
+    if (filterComments.length === 1)
+      return t('stocktake.comment-template', { filters: filterComments[0] });
+
+    const comments = t('stocktake.comment-and-template', {
+      start: filterComments.slice(0, -1).join(', '),
+      end: filterComments[filterComments.length - 1],
+    });
+
+    return t('stocktake.comment-template', { filters: comments });
   };
 
   const onSave = () => {
-    const { locationId, masterListId, itemsHaveStock, expiresBefore } =
-      createStocktakeArgs;
+    // Our API only has a `beforeOrEqualTo` filter, so just kludging the date back 1 day here
+    const adjustedExpiryDate = expiryDate
```

Comment:
Why are you taking away a day? Would just pass this straight to api since expired date is a Date field not a Datetime....

E.g. I want all stock lines that expire before 30.06.2025, so I should be getting all stock lines that expired on or before this date. 

Currently, taking away a day queries for stock lines that expire on or before 29.06.2025~ backend does a DateFilter::before_or_equal_to filter for expiresBefore

Reply:
https://github.com/msupply-foundation/open-msupply/blob/111c2c55299410a90eadc761725fe5f185d03609/server/service/src/stocktake/insert/generate.rs#L95

So yes, backend **ONLY** does a DateFilter::before_or_equal_to filter for expiresBefore. The UI says "Items expiring before" so if I choose 30/06/2025, I by that description I expect all stock that is expiring before 30/06/2025 not including the 30th. So... I shift it back a day so that the backend does <=29/06/2025 rather than <=30/06/2025.

The alternative was implementing `DateFilter::before` in the backend?

Reply:
Maybe misinterpretation? I understood that as before (date) or equal to (date) 🤔 better naming needed~